src/argowrapper/routes/routes.py

```diff
@@ -150,71 +152,51 @@ def check_team_projects_and_cohorts(fn):
+        cohort_ids = []
+        if "cohort_ids" in request_body["outcome"]:
             cohort_ids.extend(request_body["outcome"]["cohort_ids"])
 
-        if "variables" in request_body:
-            variables = request_body["variables"]
-            for v in variables:
-                if "cohort_ids" in v:
-                    cohort_ids.extend(v["cohort_ids"])
+        variables = request_body["variables"]
+        for v in variables:
+            if "cohort_ids" in v:
+                cohort_ids.extend(v["cohort_ids"])
+
+        if "source_population_cohort" in request_body:
+            cohort_ids.append(request_body["source_population_cohort"])
+
+        cohort_id_set = set(cohort_ids)
 
         if team_project and source_id and len(team_project) > 0 and len(cohort_ids) > 0:
-            header = {"Authorization": token, "cookie": "fence={}".format(token)}
-            url = "http://cohort-middleware-service/cohortdefinition-stats/by-source-id/{}/by-team-project?team-project={}".format(
-                source_id, team_project
+            # Get team project cohort ids
+            team_cohort_id_set = get_team_cohort_id(token, source_id, team_project)
+
+            logger.debug("cohort ids are " + " ".join(str(c) for c in cohort_ids))
+            logger.debug(
+                "team cohort ids are " + " ".join(str(c) for c in team_cohort_id_set)
             )
 
-            logger.info("team project is " + team_project)
-            logger.info("source_id is " + str(source_id))
-            logger.info("request url is " + url)
-
-            try:
-                r = requests.get(url=url, headers=header)
-                r.raise_for_status()
-                team_cohort_info = r.json()
-                team_cohort_id_set = set()
-                if "cohort_definitions_and_stats" in team_cohort_info:
-                    for t in team_cohort_info["cohort_definitions_and_stats"]:
-                        if "cohort_definition_id" in t:
-                            team_cohort_id_set.add(t["cohort_definition_id"])
-                cohort_id_set = set(cohort_ids)
-
-                logger.info("cohort ids are " + " ".join(str(c) for c in cohort_ids))
-                logger.info(
-                    "team cohort ids are "
-                    + " ".join(str(c) for c in team_cohort_id_set)
+            # Compare the two sets
+            if cohort_id_set.issubset(team_cohort_id_set):
+                logger.debug(
+                    "cohort ids submitted all belong to the same team project. Continue.."
+                )
+                return fn(*args, **kwargs)
+            else:
+                logger.error(
+                    "Cohort ids submitted do NOT all belong to the same team project."
+                )
+                return HTMLResponse(
+                    content="Cohort ids submitted do NOT all belong to the same team project.",
+                    status_code=HTTP_401_UNAUTHORIZED,
```

Comment:
```suggestion
                    status_code=HTTP_400_BAD_REQUEST,
```
maybe this could be a BAD REQUEST  type of error?